File: LedgerPro.Core/Entities/BankSource.cs
namespace LedgerPro.Core.Entities;

public class BankSource
{
    public Guid Id { get; set; }
    public string AccountName { get; set; } = string.Empty;
    public string AccountNumber { get; set; } = string.Empty;
    public string BankName { get; set; } = string.Empty;        
}

File: web/ledger-pro-ui/src/types/bankSource.ts
export type BankType = 'Generic' | 'NAB' | 'ANZ' | 'CBA';

export interface BankSource {
    id: string;
    bankName: string;
    accountName: string;
    accountNumber: string;
    bankType: BankType;
    generalLedgerAccountId: number;
}